src/Raven.Server/Documents/Indexes/IndexStorage.cs

```diff
@@ -209,6 +211,32 @@ public long ReadErrorsCount()
             return DateTime.FromBinary(lastIndexingTime.Reader.ReadLittleEndianInt64());
         }
 
+        public bool IsIndexInvalid(RavenTransaction tx)
```

Comment:
Better to implement this in Index and use IndexStorage.ReadStats witch is fairly cheap.